packages/packages.go

```diff
@@ -235,40 +238,103 @@ func (i *FileSystemIndexer) getPackagesFromFileSystem(ctx context.Context) (Pack
 		name    string
 		version string
 	}
+	type packageJob struct {
+		position int
+		path     string
+	}
+	numWorkers := runtime.GOMAXPROCS(0) / 2
+	mu := sync.Mutex{}
 	packagesFound := make(map[packageKey]struct{})
```

Comment:
why using here an empty struct? could we use a bool? and chech its found (by key) and true?

Reply:
Here, I just interested in whether or not the key is present. I think it is not needed to use boolean. An advantage of using`struct{}` here is that its size is also zero according to [this](https://stackoverflow.com/a/37320521).